File: backend/components/metiers/administration_publique/tests/samples/sample_test.js
// sample_test.js – Exemple ultra avancé de test unitaire clé en main (JS)

const { ApiService } = require('../api');
const { ServiceThreed } = require('../services');

describe('Sample Test Ultra Avancé – API & Service', () => {
  it('doit retourner un résultat conforme pour une action API', () => {
    const api = new ApiService({ mode: 'test' });
    const res = api.handle('PING', { foo: 'bar' });
    expect(res).toBeDefined();
    expect(res.status).toBe('OK');
  });

  it('doit auditer et logger une action service', () => {
    const service = new ServiceThreed({ mode: 'audit' });
    service.init({ version: 1 });
    const result = service.handle('ACTION', { data: 42 });
    expect(result).toHaveProperty('success', true);
    expect(service.getAuditTrail()).toContain('ACTION');
  });
});

File: backend/components/metiers/administration_publique/tests/samples/sample_test.py
# sample_test.py – Exemple ultra avancé de test unitaire clé en main (Python)
import pytest
from ..api.api import ApiService
from ..services.services import ServiceThreed

def test_api_service_ping():
    api = ApiService(options={"mode": "test"})
    res = api.handle("PING", {"foo": "bar"})
    assert res is not None
    assert res["status"] == "OK"

def test_service_audit_action():
    service = ServiceThreed(options={"mode": "audit"})
    service.init({"version": 1})
    result = service.handle("ACTION", {"data": 42})
    assert result["success"] is True
    assert "ACTION" in service.get_audit_trail()
